--- docs/js/constraints.js
+++ docs/js/constraints.js
@@ -35,6 +35,7 @@
             "samesum",
             "xaverage",
             "triplesum",
+            "windoku",
             "odd even bridge",
             "odd even count",
             "odd even sum"
@@ -360,6 +361,12 @@
             "styleset": ["", 1],
             "outside": true
         },
+        "windoku": {
+            "show": ["mo_sudoku_lb", "sub_sudoku1_lb", "sub_sudoku2_lb", "sub_sudoku3_lb", "mo_surface_lb"],
+            "modeset": ["sudoku", "surface"],
+            "submodeset": ["1", ""],
+            "styleset": ["", 1]
+        },
         "alternatingstripes": {
             "show": ["mo_sudoku_lb", "sub_sudoku1_lb", "sub_sudoku2_lb", "sub_sudoku3_lb",
                 "mo_line_lb", "sub_line2_lb", "st_line5_lb"],
